We calculated the genome-wide heterozygosity for each wolf individual above 5x using ANGSD (Korneliussen et al. 2014). We first estimated the folded site allele frequency with the realSFS program in ANGSD (-doSaf) using genotype likelihoods with the SAMtools model (-GL 1) and using the Andean fox genome to determine ancestral state. We filtered each individual bam file to keep only autosomal chromosomes, only properly paired reads (-only_proper_paired 1), excluded reads with excess of mismatches (-C 50), excluded reads with a mapping quality (-minMapQ 20), excluded reads with a depth less than 4 (-setMinDepth 4), and kept bases with a sequence quality above 20 (-minQ 20). To calculate the genome-wide heterozygosity, we divided the number of heterozygous sites by the total number of sites for each sample. ![Uploading image.pngâ€¦]()

